app/controllers/api/factions_controller.rb

```diff
@@ -0,0 +1,23 @@
+class Api::FactionsController < ApplicationController
+  def index
+    @factions = Faction.all
+
+    render json: {
+      :data => @factions.as_json(except: [:id, :created_at]),
+      :success => true,
+      :total => @factions.length,
+      :version_number => "3.0" 
+    }
+  end
+
+  def show
+    @faction = Faction.find_by(code: params[:code])
+
+    render json: {
+      :data => @faction.nil? ? [] : [@faction.as_json(except: [:id, :created_at])],
```

Comment:
Should this be just an if with the two branches instead of checking nil in each data?

Reply:
And then if v do, apply this to all the others too

Reply:
resolved by switching to the jsonapi-resource pattern.